v2 beta 6 segfault

```
package main

import (
	"log"

	"github.com/emersion/go-imap/v2"
	"github.com/emersion/go-imap/v2/imapclient"
)

func main() {
	c, err := imapclient.DialTLS("imappro.zoho.com:993", nil)
	if err != nil {
		log.Fatalf("failed to dial IMAP server: %v", err)
	}
	defer c.Close()

	if err := c.Login("root", "asdf").Wait(); err != nil {
		log.Fatalf("failed to login: %v", err)
	}

	mailboxes, err := c.List("", "%", nil).Collect()
	if err != nil {
		log.Fatalf("failed to list mailboxes: %v", err)
	}
	log.Printf("Found %v mailboxes", len(mailboxes))
	for _, mbox := range mailboxes {
		log.Printf(" - %v", mbox.Mailbox)
	}

	selectedMbox, err := c.Select("INBOX", nil).Wait()
	if err != nil {
		log.Fatalf("failed to select INBOX: %v", err)
	}
	log.Printf("INBOX contains %v messages", selectedMbox.NumMessages)

	if selectedMbox.NumMessages > 0 {
		seqSet := imap.SeqSetNum(1)
		fetchOptions := &imap.FetchOptions{Envelope: true}
		messages, err := c.Fetch(seqSet, fetchOptions).Collect()
		if err != nil {
			log.Fatalf("failed to fetch first message in INBOX: %v", err)
		}
		log.Printf("subject of first message in INBOX: %v", messages[0].Envelope.Subject)
	}

	if err := c.Logout().Wait(); err != nil {
		log.Fatalf("failed to logout: %v", err)
	}
}
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5bfddf]

goroutine 1 [running]:
github.com/emersion/go-imap/v2/imapclient.(*Options).dialer(...)
        /home/khalid/go/pkg/mod/github.com/emersion/go-imap/v2@v2.0.0-beta.6/imapclient/client.go:124
github.com/emersion/go-imap/v2/imapclient.DialTLS({0x6a464b, 0x14}, 0x0)
        /home/khalid/go/pkg/mod/github.com/emersion/go-imap/v2@v2.0.0-beta.6/imapclient/client.go:237 +0xbf
main.main()
        /home/khalid/projects/golang/imap2/main.go:11 +0x48
exit status 2
```

However beta 5 is working fine.

--- a/imapclient/client.go
+++ b/imapclient/client.go
@@ -220,6 +220,10 @@ func NewStartTLS(conn net.Conn, options *Options) (*Client, error) {
 
 // DialInsecure connects to an IMAP server without any encryption at all.
 func DialInsecure(address string, options *Options) (*Client, error) {
+	if options == nil {
+		options = &Options{}
+	}
+
 	conn, err := options.dialer().Dial("tcp", address)
 	if err != nil {
 		return nil, err
@@ -229,6 +233,10 @@ func DialInsecure(address string, options *Options) (*Client, error) {
 
 // DialTLS connects to an IMAP server with implicit TLS.
 func DialTLS(address string, options *Options) (*Client, error) {
+	if options == nil {
+		options = &Options{}
+	}
+
 	tlsConfig := options.tlsConfig()
 	if tlsConfig.NextProtos == nil {
 		tlsConfig.NextProtos = []string{"imap"}

